### PR TITLE
Add unit to Instrument selection criteria

### LINKED ISSUE #3101: Support units in Instrument selector in Metric View
**What are you trying to achieve?**
I would like to have the ability to say for a view, in its instrument selector: "Select all histogram type, which has the units 'seconds'". 
This will allow me to say: All histograms with milliseconds unit with name matching "pulsar.messaging.*" should have the following buckets - 1, 5, 10, 20, 100, 1000, 5000. 
Another example: All histograms with bytes unit should have the following buckets: 100 bytes, 500 bytes, 1kb, 2kb, and 10kb.

**What did you expect to see?**
In the [section](https://opentelemetry.io/docs/reference/specification/metrics/sdk/#view) describing the instrumentation selector for a view, starting with "The Instrument selection criteria (required), which covers:" another bullet show be added saying "The `unit` of the instrument (optional)".

**Additional context.**

In my opinion, the ability to specify a bucket list based on units (seconds, bytes, milliseconds) combined with instrumentation scope or wildcard on a metric name is a very powerful and useful feature to have. Without it, and without hints API, one finds themself creating so many views, one for each histogram defined or trying to devise a naming scheme that would be matched using the name wildcard.

### PR DESCRIPTION
**Pull request by asafm**
## Changes

Added unit criteria to the instrument selection criteria of a view.

Please provide a brief description of the changes here.

For non-trivial changes, follow the [change proposal process](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CONTRIBUTING.md#proposing-a-change) and link to the related issue(s) and/or [OTEP(s)](https://github.com/open-telemetry/oteps), update the [`CHANGELOG.md`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CHANGELOG.md), and also be sure to update [`spec-compliance-matrix.md`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md) if necessary.

## Related issues 

Fixes #3101 

## Related [OTEP(s)](https://github.com/open-telemetry/oteps) 
none

### CODE REVIEW

**Review by MrAlias — APPROVED** [review]

**Comment by MrAlias** [comment]
@asafm please sign the CLA so these changes can be considered for merging.

**Review by reyang — APPROVED** [review]
LGTM with a changelog entry + clear CLA.

File: specification/metrics/sdk.md [hunk 1]
```diff
@@ -179,6 +179,7 @@ are the inputs:
     matching zero or more characters.  If wildcards are not supported in general,
     OpenTelemetry SDKs MUST specifically recognize the single `*` wildcard
     as matching all instruments.
+  * The `unit` of the Instrument(s) (optional).
```
Comment (jsuereth) [inline specification/metrics/sdk.md:182]: Can you make sure this is also a row in the compatibility matrix markdown file?

**Review by jsuereth — APPROVED** [review]

File: specification/metrics/sdk.md (hunk 1 above)
Comment (asafm) [reply specification/metrics/sdk.md:182]: @jsuereth I added - please tell me if that's ok?

**Comment by asafm** [comment]
@reyang Added line to changelog - I linked the PR, wasn't sure if I'm suppose to link the PR or the issue

**Comment by cijothomas** [comment]
@asafm The CLA check is missing. Could you add that ?

**Comment by asafm** [comment]
> @asafm The CLA check is missing. Could you add that ?

Working my company to get them to sign and add me

File: specification/metrics/sdk.md (hunk 1 above)
Comment (arminru) [reply specification/metrics/sdk.md:182]: @asafm looks good, thank you!

**Review by jack-berg — APPROVED** [review]

**Comment by asafm** [comment]
@cijothomas CLA approved

File: spec-compliance-matrix.md [hunk 2]
```diff
@@ -95,117 +95,118 @@ formats is required. Implementing more than one format is optional.
...
+| The `View` instrument selection criteria is as specified.                                                                                                              |          |    | +    | +   | +      |      |        | +   |      | +   | +    |       |
+| The `View` instrument selection criteria supports wildcards.                                                                                                           | X        |    | +    | +   | +      |      |        |     |      | +   | +    |       |
+| The `View` instrument selection criteria supports the match-all wildcard.                                                                                              |          |    | +    | +   | +      |      |        |     |      | +   | +    |       |
+| The `View` instrument selection criteria supports a unit.                                                                                                              |          | +  |      |     |        |      |        |     |      |     |      |       |
```
Comment (cijothomas) [inline spec-compliance-matrix.md:169]: we do not have an entry for other things like name,  type, meter name, version etc, so I do not think we should add one for "unit".  This compliance list is already too big for metrics, making it hard to follow sometimes.

"The `View` instrument selection criteria is as specified." -- this existing entry should be good enough in my opinion.

**Review by cijothomas — APPROVED** [review]
LGTM.
left a non-blocking comment about the compliance matrix getting too big, so suggest avoid adding an entry there.

File: spec-compliance-matrix.md (hunk 2 above)
Comment (asafm) [reply spec-compliance-matrix.md:169]: @jsuereth Can you respond here? You asked for it in the comments. WDYT?

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
cc @jmacd

File: spec-compliance-matrix.md (hunk 2 above)
Comment (asafm) [reply spec-compliance-matrix.md:169]: @cijothomas I thought the whole idea of this list is to have some sort of TO-DO for each SDK. If we have added unit support, should this mark as a TO-DO for the SDKs thus add a line for it?

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please rebase @asafm

**Comment by asafm** [comment]
@bogdandrutu done
